src/documentmanager/plugin.ts

```diff
@@ -59,180 +47,184 @@ export
+
+  app.commands.add([
+    {
+      id: newNotebookId,
+      handler: () => {
+        browser.newUntitled('notebook').then(
+          contents => manager.open(contents)
+        );
+      }
+    }
+  ]);
+  app.palette.add([
+    {
+      command: newNotebookId,
+      category: 'File Operations',
+      text: 'New Notebook',
+      caption: 'Create a new Jupyter Notebook'
+    }
+  ]);
 
+  // Add the command for saving a document.
+  let saveDocumentId = 'file-operations:save';
 
-export
-function register(container: Container): void {
-  container.register(IDocumentManager, {
-    requires: [IAppShell, IFileBrowserWidget],
-    create: (appShell: IAppShell, browser: IFileBrowserWidget): IDocumentManager => {
-      let manager = new DocumentManager();
-      let id = 0;
-      browser.openRequested.connect((browser, model) => {
-        let widget = manager.open(model);
-        if (!widget.id) widget.id = `document-manager-${++id}`;
-        if (!widget.isAttached) appShell.addToMainArea(widget);
-        let stack = widget.parent;
-        if (!stack) {
-          return;
-        }
-        let tabs = stack.parent;
-        if (tabs instanceof TabPanel) {
-          tabs.currentWidget = widget;
-        }
-      });
-      return manager;
+  app.commands.add([
+    {
+      id: saveDocumentId,
+      handler: () => {
+        manager.save();
+        return true;
+      }
+    }
+  ]);
+  app.palette.add([
+    {
+      command: saveDocumentId,
+      category: 'File Operations',
+      text: 'Save Document',
+      caption: 'Save the current document'
+    }
+  ]);
+
+  // Add the command for reverting a document.
+  let revertDocumentId = 'file-operations:revert';
+
+  app.commands.add([
+    {
+      id: revertDocumentId,
+      handler: () => {
+        manager.revert();
+        return true;
+      }
+    }
+  ]);
+  app.palette.add([
+    {
+      command: revertDocumentId,
+      category: 'File Operations',
+      text: 'Revert Document',
+      caption: 'Revert the current document'
+    }
+  ]);
+
+  // Add the command for closing a document.
+  let closeDocumentId = 'file-operations:close';
+
+  app.commands.add([
+    {
+      id: closeDocumentId,
+      handler: () => {
+        manager.close();
+        return true;
+      }
+    }
+  ]);
+  app.palette.add([
+    {
+      command: closeDocumentId,
+      category: 'File Operations',
+      text: 'Close Document',
+      caption: 'Close the current document'
+    }
+  ]);
+
+  // Add the command for closing all documents.
+  let closeAllId = 'file-operations:close-all';
+
+  app.commands.add([
+    {
+      id: closeAllId,
+      handler: () => {
+        manager.closeAll();
+        return true;
+      }
+    }
+  ]);
+  app.palette.add([
+    {
+      command: closeAllId,
+      category: 'File Operations',
+      text: 'Close All',
+      caption: 'Close all open documents'
+    }
+  ]);
+
+  browser.widgetFactory = model => {
```

Comment:
move the browser to a separate extension which requires the document manager. That extension can connect to the fb signal and call open on the manager and add the widget to the dom where it wants.